src/parsers/perf/perfparser.cpp

```diff
@@ -1862,6 +1864,8 @@ void PerfParser::filterResults(const Data::FilterAction& filter)
         emit tracepointDataAvailable(tracepointResults);
         emit eventsAvailable(events);
         emit parsingFinished();
+
+        m_costAggregationChanged = false;
```

Comment:
dito, this can race

I suggest you instead add a signal with the last-used cost aggregation, then store that in a member and instead of the bool flag you then compare if the last-used cost aggregation is different to what is now requested